tsdb/engine/tsm1/file_store.go

```diff
@@ -1570,24 +1568,38 @@ type purger struct {
 }
 
 func (p *purger) add(files []TSMFile) {
+	var fileNames []string
 	p.mu.Lock()
 	for _, f := range files {
-		p.files[f.Path()] = f
+		fileName := f.Path()
+		fileNames = append(fileNames, fileName)
+		p.files[fileName] = f
 	}
 	p.mu.Unlock()
-	p.purge()
+	p.purge(fileNames)
 }
 
-func (p *purger) purge() {
+func (p *purger) purge(fileNames []string) {
+	logger, logEndOp := logger.NewOperation(p.logger, "Purge held files", "filestore_purger")
+
+	logger.Info("added", zap.Int("count", len(fileNames)))
+	logger.Debug("purging", zap.Strings("files", fileNames))
 	p.mu.Lock()
 	if p.running {
 		p.mu.Unlock()
+		logger.Info("already running, files added to previous operation")
+		logEndOp()
 		return
 	}
 	p.running = true
 	p.mu.Unlock()
 
 	go func() {
+		var purgeCount int
+		defer func() {
+			logger.Info("removed", zap.Int("files", purgeCount))
```

Comment:
Would it be a good idea to log out the file names that were removed as well as the count, I see we log the names out above, this may make it easier to compare the lists in the logs if any issues arise.

Reply:
We print them out as they are deleted when the log level is `debug`